pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2021-2022 Carli Freudenberg <carli.freudenberg@energymeteo.de>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+"""All alphanumeric unicode character are allowed in Python but due
+to similarities in how they look they can be confused.
+
+See: https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers
+
+The following checkers are intended to make users are aware of these issues.
+"""
+
+import re
+import sys
+from typing import Optional, Union
+
+from astroid import nodes
+
+import pylint.checkers.base
+import pylint.checkers.utils
+from pylint import interfaces
+from pylint.constants import HUMAN_READABLE_TYPES
+from pylint.lint import PyLinter
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+
+if sys.version_info[:2] >= (3, 7):
+    # pylint: disable-next=fixme
+    # TODO: Remove after 3.6 has been deprecated
+    Py37Str = str
+else:
+
+    class Py37Str(str):
+        # Allow Python 3.6 compatibility
+        def isascii(self: str) -> bool:
+            return all("\u0000" <= x <= "\u007F" for x in self)
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NonAsciiNamesChecker(
+    pylint.checkers.BaseChecker, pylint.checkers.base.NameCheckerHelper
+):
+    """A strict name checker only allowing ASCII
+
+    If your programming guideline defines that you are programming in English,
+    then there should be no need for non ASCII characters in Python Names.
+    Everybody else can simply not use this check.
+
+    Note: This check only checks Names, so it ignores the content of
+          docstrings and comments!
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    priority = -1
+
+    msgs = {
+        "C2401": (
+            '%s name "%s" contains a non-ASCII character, rename it.',
+            "non-ascii-identifier",
+            (
+                "Used when the name contains at least one non-ASCII unicode character."
+                "See https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers"
+                " for a background why this could be bad."
+            ),
+            {"old_names": [("C0144", "non-ascii-name")]},
```

Comment:
```suggestion
            "non-ascii-name",
            (
                "Used when the name contains at least one non-ASCII unicode character."
                "See https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers"
                " for a background why this could be bad."
            ),
            {"old_names": [("C0144", "old-non-ascii-name")]},
```
We can rename the msgid and not the symbol with this one simple trick (and avoid the new symbol that is just a synonym altogether)

Reply:
I kind of like *identifier* more and think it is more fitting as the *name*.
Name is more generic IMO. Also considering that now a file__name__ is an extra message.

@DanielNoord and @Pierre-Sassoulas could you agree about this. Because changing it back would mean not renaming some files but instead adopting the messages everywhere again (okay not that much of work, that to `sed`).

Just need to know this before going forward with some issues mentioned here.

Reply:
Kept the old name.